src/test/java/com/agileavengers/icuconnectbackend/service/ICommunityServiceTest.java

```diff
@@ -489,4 +505,79 @@ void getCommunityRatingAverage() {
 
     }
 
-}
+    @Test
+    void createCommunityPost() {
+        Instructor instructor = Instructor.builder().id(10L).name("Test Instructor").build();
+        Community community = Community.builder().id(1L).name("Test Community").instructor(instructor).moduleId("UZH1234").build();
+        when(communityRepository.findCommunityByModuleId(community.getModuleId()))
+                .thenAnswer(i -> Optional.of(community));
+
+
+        User user1 = User.builder().username("Test1").password("anything").id(2L).subscriptionSet(Set.of(community)).build();
+
+        PostDto post = PostDto.builder().id(2L).title("Question 1 Title").text("Question 1 Text")
+        .build();
+
+        when(userRepository.findByUsername(user1.getUsername())).thenAnswer(i -> Optional.of(user1));
+
+        when(postRepository.save(Mockito.any(Post.class)))
+            .thenAnswer(i -> {
+                Post argument = (Post) i.getArguments()[0];
+                argument.setId(1L);
+                argument.setCommunity(community);
```

Comment:
Is it needed to set these fields that are not ID?

Reply:
You are right, it works without. Removed.